src/File.hack

```diff
@@ -22,6 +22,45 @@ final class File {
     return $this->isDirty;
   }
 
+  private function getLines(): vec<string> {
+    return Str\split($this->contents, "\n");
+  }
+
+  const type TLineNumber1Base = int;
+
+  /**
+   * Returns the 1-base line number of lint markers in the file.
+   */
+  <<__Memoize>>
+  public function lintMarkers(): dict<string, keyset<self::TLineNumber1Base>> {
```

Comment:
lintMarkersForLineBasedSuppression

Not all linters should respect these, especially AST-based linters (they should require that the suppression be attached to the specific AST node).

That's probably the best for hh-client based linters; given ASTs are cached, it shoudln't be too wasteful. The position can be transformed to a node via HHAST\find_node_at_position() - and the AST identifies which comments correspond to which nodes.

if memoized, this does need sharing between linters, but it's not great for separation of concerns :(

Reply:
I am not sure if HHVM support something similar to ES6 `WeakMap` to implement a non-intrusive cache.

Reply:
It's not worth to introduce some mechanism of Dependency Injection in this simple library. 
We might be able to implement the [Cake Pattern](https://www.baeldung.com/scala/cake-pattern) here, but Hack `trait`s are more restrictive than Scala `trait`s. 

Maybe there is no simple solution for separation of concerns here.